x/ibc/04-channel/handler.go

```diff
@@ -0,0 +1,181 @@
+package ics04
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/ibc/04-channel/keeper"
+	"github.com/cosmos/cosmos-sdk/x/ibc/04-channel/types"
+)
+
+// NewHandler creates a new Handler instance for IBC connection
+// transactions
+func NewHandler(k keeper.Keeper) sdk.Handler {
+	return func(ctx sdk.Context, msg sdk.Msg) sdk.Result {
```

Comment:
Same comment as in previous ICS implementations about internal / external modules, but up to the SDK team.

Reply:
I actually had the same question about the internal structure when I reviewed the first implementations of the PRs... Ideally, we should structure the module in a way that every ICS implementation is clearly distinguishable for ease of auditing

Reply:
I believe @jackzampolin had the same question on several of his reviews. Let's discuss it today on the call